### PR TITLE
64 vmem

### PR DESCRIPTION
**Pull request by GregAC**
I considered leaving the existing 32-bit .vmem without a naming change but felt it might just confuse things so better to have the pain now of changing to .32.vmem.

I did a grep through the repository for .vmem and fixed up everything I could find which wasn't a huge amount anyway.

### CODE REVIEW

**Comment by GregAC** [comment]
Oh and if you're wondering why we need this see #1941, it's possible the structure of the .64.vmem needs adjusting but that's easy enough to do.

File: util/embedded_target.py [hunk 1]
```diff
@@ -100,7 +100,8 @@ def main():
 
     bin_file = run_objcopy(args.objcopy, args.input, args.basename, args.outdir)
     run_objdump(args.objdump, args.input, args.basename, args.outdir)
-    run_srec_cat(args.srec_cat, bin_file, args.basename, args.outdir)
+    run_srec_cat(args.srec_cat, bin_file, args.basename, args.outdir, 4)
+    run_srec_cat(args.srec_cat, bin_file, args.basename, args.outdir, 8)
```
Comment (imphil) [inline util/embedded_target.py:104]: Can you use a named argument for the last element, `width_byte = 8`?

File: hw/dv/tools/rules.mk [hunk 2]
```diff
@@ -47,10 +47,10 @@ ifneq (${SW_NAME},)
...
 
 	mkdir -p $(SW_BUILD_DIR)/sw $(SW_BUILD_DIR)/rom
-	cp $(SW_BUILD_DIR)/build-out/sw/device/boot_rom/boot_rom_$(SW_BUILD_DEVICE).vmem \
-		$(SW_BUILD_DIR)/rom/rom.vmem
-	cp $(SW_BUILD_DIR)/build-out/sw/device/$(SW_DIR)/$(SW_NAME)_$(SW_BUILD_DEVICE).vmem \
-		$(SW_BUILD_DIR)/sw/sw.vmem
+	cp $(SW_BUILD_DIR)/build-out/sw/device/boot_rom/boot_rom_$(SW_BUILD_DEVICE).32.vmem \
+		$(SW_BUILD_DIR)/rom/rom.32.vmem
+	cp $(SW_BUILD_DIR)/build-out/sw/device/$(SW_DIR)/$(SW_NAME)_$(SW_BUILD_DEVICE).32.vmem \
+		$(SW_BUILD_DIR)/sw/sw.32.vmem
```
Comment (sriyerg) [inline hw/dv/tools/rules.mk:53]: You can revert the changes in this file. It is a part of the old Makefile flow which is already broken in other places. We do not plan to support it going forward. It will be removed eventually.

File: util/embedded_target.py [hunk 3]
```diff
@@ -33,30 +33,30 @@ def run_objcopy(objcopy, input, basename, outdir):
...
         '--output-target', 'binary',
         input, output,
     ]
     subprocess.run(cmd, check=True)
     return output
 
-
-def run_srec_cat(srec_cat, input, basename, outdir):
-    # TODO: Replace command for objcopy once this bug is fixed and released.
+def run_srec_cat(srec_cat, input, basename, outdir, byte_width):
```
[the remark this answers is not in the document]
Comment (GregAC) [reply util/embedded_target.py:47]: I left it was `byte_width` in the end, I felt both `width_byte` and `width_bytes` didn't read quite as well it's more obvious what `byte_width` refers to.

**Comment by GregAC** [comment]
> Ci very likely fails due to https://github.com/lowRISC/opentitan/blob/master/azure-pipelines.yml#L263, you need to adjust the bootrom vmem file in the pipeline config.

Ah whoops the one file I forgot to adjust after seeing it appear in the search!

I also pushed embedded_target.py through ./util/lintpy.py which resulted in the other formatting changes you see.

File: util/embedded_target.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sriyerg) [reply util/embedded_target.py:47]: `word_size`?

[the remark this answers is not in the document]
Comment (imphil) [reply util/embedded_target.py:47]: Something which includes the unit is really helpful here given we use both in various places.

File: util/embedded_target.py [hunk 4]
```diff
@@ -68,39 +84,41 @@ def main():
...
+                        required=True,
+                        help='Absolute path to objdump tool.')
+    parser.add_argument('--outdir',
+                        '-d',
+                        required=True,
+                        help="Output directory.")
+    parser.add_argument('--srec_cat',
+                        '-s',
+                        required=True,
+                        help='Absolute path to srec_cat tool.')
```
Comment (imphil) [inline util/embedded_target.py:103]: Yapf doesn't give stable output for these argument lists, please avoid adding them to the diff if not needed.

File: util/embedded_target.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (tjaychen) [reply util/embedded_target.py:47]: i agree with @sriyerg , `word_size` or `num_bytes` feels much more representative.  In the flash case, number of the words per row, bits per byte are all consistent, they just grew the number of bytes per word. 

As a side question, when there is eventually parity / ECC, how do you envision this changing? Would there be another python function that processes the run_srec_cat output to append on more bits?

[the remark this answers is not in the document]
Comment (imphil) [reply util/embedded_target.py:47]: On ECC/Parity: Yes, but it really depends on what kind of information we want to add and how it is calculated. But let's continue this discussion in https://github.com/lowRISC/opentitan/issues/1941.
